### ISSUE TITLE
LogNormalFactory cannot build with given gamma

### ISSUE THREAD

**Issue by mbaudin47** [openturns/openturns#1727]
I have a sample and I want to fit a `LogNormal` distribution using method of least squares. The algorithm uses a non linear least squares algorithm to estimate gamma, which is quite costly. However, I know that if gamma is set, then the method uses linear least squares, which is instantaneous and provides actually a better estimate. 
But the script:
```
import openturns as ot
sample = ot.LogNormal().getSample(100)
gamma = 0.0
factory = ot.LogNormalFactory()
distribution = factory.buildMethodOfLeastSquares(sample, gamma)
```
fails:
```
TypeError: buildMethodOfLeastSquares() takes 2 positional arguments but 3 were given
```
This is because the constructor is private, unfortunately:

https://github.com/openturns/openturns/blob/bc1ba291d2377bdd2bbefce9009f1790e3443305/lib/src/Uncertainty/Distribution/openturns/LogNormalFactory.hxx#L71

This does not match the doc, which is unclear about the status of gamma, which not present in the inputs arguments but is still described:
![image](https://user-images.githubusercontent.com/31351465/105611946-e3ac0800-5db8-11eb-93e3-8ff46ccafdf7.png)

I wish I had two constructors:
* `buildMethodOfLeastSquares(sample)` when I want to estimate gamma with non linear least squares
* `buildMethodOfLeastSquares(sample, gamma)` when gamma is known (perhaps estimated with another method)

**Comment by regislebrun** [comment]
Hi @mbaudin47 ,
There is currently no generic parameter management in distributions and distribution factories, except for the generic ones: MethodOfMomentsFactory, MaximumLikelihoodFactory and LeastSquaresDistributionFactory. You should use one of them waiting for the implementation of the parameter management we still have to define:
```
import openturns as ot
sample = ot.LogNormal().getSample(100)
gamma = 0.0
factory = ot.LeastSquaresDistributionFactory(ot.LogNormal())
factory.setKnownParameter([gamma], [2])
distribution = factory.build(sample)
print(distribution)
```
with the result:
```
LogNormal(muLog = 0.0504315, sigmaLog = 1.03219, gamma = 0)
```

Cheers

Régis

**Comment by mbaudin47** [comment]
Good to have news from you. I did not think of this way: thank you for the advice. However, notice that `LeastSquaresDistributionFactory` does not behave the way `LogNormalFactory().buildMethodOfLeastSquares()` does:
* `LeastSquaresDistributionFactory` uses the CDF differences,
* `LogNormalFactory().buildMethodOfLeastSquares()` uses the inverse of the Gaussian CDF of the empirical CDF.

For a Log-Normal distribution, the second option is more appropriate. Perhaps an example would better show how this specific transformation maps to the Log-Normal distribution.

**Comment by regislebrun** [comment]
Right, I have been fouled by the includes in LogNormalFactory.cxx: there is a
```
#include "openturns/LeastSquaresDistributionFactory.hxx"
```
but it is unused, and the least-squares problem solved by a (much more) twisted algorithm, where (in my opinion) way too much linear least-squares problems are solved!
